src/main/java/org/springframework/integration/aws/outbound/SnsMessageHandler.java

```diff
@@ -172,6 +196,30 @@ protected Future<?> handleMessageToAws(Message<?> message) {
 				publishRequest.setSubject(subject);
 			}
 
+			if (this.messageGroupIdExpression != null) {
+				if (!topicArn.endsWith(".fifo")) {
+					if (logger.isWarnEnabled()) {
+						logger.warn(String.format(
```

Comment:
Let's see if we can leverage the mentioned `LogAccessor` API:
```
warn(Supplier<? extends CharSequence> messageSupplier)
```
and supply a log message lazily via `LogMessage format(String format, Object arg1)`

Reply:
I was copying off of `AbstractMessageAttributesHeaderMapper` (since I knew that `SqsHeaderMapper` logs!), which depends directly on the Commons logger.

I tried using a lambda, but `topicArn` isn't effectively final because of the reassignment if there's a `resourceIdResolver`; however, `LogMessage` is apparently intended to be used directly as it's lazily evaluated.